Modules/Managers/SpawnManager.cs

```diff
@@ -0,0 +1,116 @@
+
+using CounterStrikeSharp.API.Core;
+using CounterStrikeSharp.API.Modules.Utils;
+using RetakesPlugin.Modules.Configs;
+using RetakesPlugin.Modules.Enums;
+
+namespace RetakesPlugin.Modules.Managers;
+
+public class SpawnManager
+{
+    private readonly Translator _translator;
+    private readonly MapConfig _mapConfig;
+	private readonly Dictionary<Bombsite, Dictionary<CsTeam, List<Spawn>>> _spawns = new();
+	private readonly bool _spawnsInitialized = false;
+
+    public SpawnManager(Translator translator, MapConfig mapConfig)
+    {
+        _translator = translator;
+		_mapConfig = mapConfig;
+
+		foreach (var spawn in _mapConfig.GetSpawnsClone())
+		{
+			if (!_spawns.ContainsKey(spawn.Bombsite))
+			{
+				_spawns.Add(spawn.Bombsite, new Dictionary<CsTeam, List<Spawn>>());
+			}
+
+			if (!_spawns[spawn.Bombsite].ContainsKey(spawn.Team))
+			{
+				_spawns[spawn.Bombsite].Add(spawn.Team, new List<Spawn>());
+			}
+
+			_spawns[spawn.Bombsite][spawn.Team].Add(spawn);
+		}
+		_spawnsInitialized = true;
+    }
+
+	public List<Spawn> GetSpawns(Bombsite bombsite, CsTeam? team = null)
+	{
+		if (!_spawnsInitialized)
+		{
+			return new List<Spawn>();
+		}
+
+		if (team == null)
+		{
+			return _spawns[bombsite].SelectMany(entry => entry.Value).ToList();
+		}
+
+		return _spawns[bombsite][(CsTeam)team];
+	}
+
+	/**
+     * This function returns a the player who should be the planter and moves all players to random spawns based on bomb site.
+     */
+	public CCSPlayerController? HandleRoundSpawns(Bombsite bombsite, HashSet<CCSPlayerController> players)
```

Comment:
We may also want to add some sanity checks in here to ensure that there are enough spawns for players etc / if we run out of spawns to allocate then we should throw an exception